base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/dialogs/ImportComponentDialog.java

```diff
@@ -0,0 +1,215 @@
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Text;
+
+import uk.ac.stfc.isis.ibex.configserver.configuration.Configuration;
+import uk.ac.stfc.isis.ibex.instrument.Instrument;
+import uk.ac.stfc.isis.ibex.instrument.InstrumentInfo;
+import uk.ac.stfc.isis.ibex.ui.configserver.ImportVariables;
+import uk.ac.stfc.isis.ibex.ui.configserver.editing.components.ComponentTable;
+import uk.ac.stfc.isis.ibex.ui.mainmenu.instrument.InstrumentTable;
+
+/**
+ * Dialog for importing a component.
+ */
+public class ImportComponentDialog extends TitleAreaDialog  {
+	
+	private static final int WIDTH = 800;
+	private static final int HEIGHT = 800;
+	
+	private final String title;
+	private final String subTitle;
+	private InstrumentTable instrumentTable;
+	private ComponentTable componentTable;
+	private Button importButton;
+	
+	private ImportVariables importVariables;
+	private Configuration selectedComponent = null;
+
+	
+	/**
+     * Constructor.
+     * 
+     * @param parentShell 		parent shell to run dialogue
+	 * @param title         	title of dialogue
+	 * @param subTitle      	action being taken
+	 * @param importVariables	variables used for importing
+     */
+	public ImportComponentDialog(Shell parentShell, String title, String subTitle, ImportVariables importVariables) {
+		super(parentShell);
+		this.title = title;
+		this.subTitle = subTitle;
+		this.importVariables = importVariables;
+	}
+	
+	@Override
+	protected void configureShell(Shell shell) {
+		super.configureShell(shell);
+		shell.setText(title);
+	}
+	
+	@Override
+	protected Point getInitialSize() {
+        return new Point(WIDTH, HEIGHT);
+	}
+	
+	@Override
+	protected Control createDialogArea(Composite parent) {
+        setTitle(subTitle);
+		Composite container = new Composite(parent, SWT.FILL);
+		container.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 1, 1));
+		container.setLayout(new GridLayout(2, true));
+		
+		createIntrumentGroup(container);
+		createComponentGroup(container);
+		
+		importVariables.addUiThreadPropertyChangeListener("components", new PropertyChangeListener() {
+			@Override
+			public void propertyChange(PropertyChangeEvent evt) {
+				componentTable.setComponents(importVariables.getComponents());
+			}
+		});
+		
+		importVariables.addUiThreadPropertyChangeListener("status", new PropertyChangeListener() {
+			@Override
+			public void propertyChange(PropertyChangeEvent evt) {
+				componentTable.enabled(importVariables.connected() && importVariables.versionMatch());
+				
+				if (!importVariables.connected()) {
+					setErrorMessage("Selected instrument's Block Server is offline.");
+				} else if (!importVariables.versionMatch()) {
+					setErrorMessage("Selected instrument is on a different version.");
+				} else {
+					setErrorMessage(null);
+				}
```

Comment:
This sort of thing is really "UI logic" that ideally belongs in a viewmodel, not a view. Can you have a look for other similar logic and refactor out into viewmodels wherever appropriate?